libs/natlint/src/main/java/org/amshove/natlint/analyzers/DefineDataParameterAnalyzer.java

```diff
@@ -0,0 +1,64 @@
+package org.amshove.natlint.analyzers;
+
+import org.amshove.natlint.api.AbstractAnalyzer;
+import org.amshove.natlint.api.DiagnosticDescription;
+import org.amshove.natlint.api.IAnalyzeContext;
+import org.amshove.natlint.api.ILinterContext;
+import org.amshove.natparse.DiagnosticSeverity;
+import org.amshove.natparse.NodeUtil;
+import org.amshove.natparse.ReadOnlyList;
+import org.amshove.natparse.natural.ISyntaxNode;
+import org.amshove.natparse.natural.IScopeNode;
+
+public class DefineDataParameterAnalyzer extends AbstractAnalyzer
+{
+	public static final DiagnosticDescription USE_OF_INLINE_PARAMETER_IS_DISCOURAGED = DiagnosticDescription.create(
```

Comment:
The message and the name here says discouraged while you renamed the editorconfig stuff to disallow :-)

I'm more in favor of discourage, as the severity is warning and not error, so lets change the editorconfig stuff back to `discourage_`

Reply:
I did that because we have other analysers that use "disallow", so I thought it'd be more consistant to use "disallow". Let me know if you still wanna change it back. :)

Reply:
I'm not really sure about the naming. `Disallowed` sounds like something breaks/stops if you do it, but we don't control any compile process or something like that.

Having a warning or info severity saying something is disallowed also seems strange, thats why I went with discouraged.

I think we can go with whatever message, but should be consistent with `.editoroconfig` stuff. Changing the config keys is also a breaking change, because as a user you could already have enabled NL028 and NL030 and therefore not have it enabled anymore when you update to the next version.

My suggestion is to stick with `discourage_*` config names, but let the implementer (you in this case) decide what the message will say. A user can always overwrite the severity, so the message can always feel unfit :-)